sympy/sets/sets.py

```diff
@@ -381,6 +381,25 @@ def _contains(self, other):
         """
         raise NotImplementedError(f"{type(self).__name__}._contains")
 
+    def convert_to_interval(set):
+        if isinstance(set, Interval):
+            return set
+        elif isinstance(set, Union):
+            intervals = [i for i in set.args if isinstance(i, Interval)]
+            if intervals:
+                lower_bound = Min([i.args[0] for i in intervals])
+                upper_bound = Max([i.args[1] for i in intervals])
+                return Interval(lower_bound, upper_bound)
+        elif isinstance(set, Intersection):
+            # Simplify the Intersection first
+            set = set.simplify()
```

Comment:
The `simplify` function should not be used in library code. This is potentially a very expensive function. A query like `is_subset` is supposed to be a fast check that handles simple cases so it should definitely not use `simplify`.

Reply:
Thanks for your answer, I have found the existing problem in my code, and I will try to solve this problem again. Will adding the range of user defined unknowns solve this issue?